common/src/main/java/com/wynntils/features/user/redirects/TerritoryMessageRedirectFeature.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user.redirects;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureCategory;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.notifications.NotificationManager;
+import com.wynntils.mc.event.SubtitleSetTextEvent;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@FeatureInfo(category = FeatureCategory.REDIRECTS)
+public class TerritoryMessageRedirectFeature extends UserFeature {
+    private static final Pattern TERRITORY_MESSAGE_PATTERN = Pattern.compile("§7\\[You are now (\\S+) (.+)\\]");
+
+    // Handles the subtitle text event.
+    @SubscribeEvent
+    public void onSubtitleSetText(SubtitleSetTextEvent event) {
+        String codedString = ComponentUtils.getCoded(event.getComponent());
+        Matcher matcher = TERRITORY_MESSAGE_PATTERN.matcher(codedString);
+        if (!matcher.matches()) return;
+
+        event.setCanceled(true);
+
+        String rawDirection = matcher.group(1);
+        String rawTerritoryName = matcher.group(2);
+        String directionalArrow;
+
+        switch (rawDirection) {
+            case "entering":
+                directionalArrow = "→";
+                break;
+            case "leaving":
+                directionalArrow = "←";
+                break;
+            default:
+                return;
+        }
+
+        // Want to account for weird stuff like "the Forgery" and make it "The Forgery"
+        // for the sake of our brief message (looks odd otherwise).
+        String territoryName = rawTerritoryName.substring(0, 1).toUpperCase() + rawTerritoryName.substring(1);
```

Comment:
Is there not already a utility function to do this? Either in our classes or like google CaseFormat. If not, consider making this a string util function.

Reply:
it was in apache stringutils, whoops